Automattic/jetpack
# Add initial state to the connection package

Note: This PR was originally drafted and discussed in #21745

This PR proposes a way for the Connection composer package to expose an Initial State that can be consumed by the connection JS package. This will make it easier for consumers to use the connection js package and not have to worry about its initial state.

For now this is working for the ConnectionStatus, but it can also work for the `apiRoot` and `registrationNonce`.

#### Changes proposed in this Pull Request:

* Add initial state to the connection package

#### Jetpack product discussion


https://github.com/Automattic/jetpack/pull/21745

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Build Jetpack and Backup plugins
* Disconnect
* Visit both plugins pages and make sure the request to `v4/connection` endpoint is NOT performed.
* Go to `class-jetpack-backup.php` in the Backup plugin and comment out the line that adds the Connection Initial state. Refresh the page and see that now the request is performed. You can do the same with Jetpack
